exercises/practice/yacht/.meta/example.c

```diff
@@ -0,0 +1,108 @@
+#include "yacht.h"
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+static int number_of_x(dice_t dice, int x)
+{
+   int sum = 0;
+   for (int i = 0; i < 5; ++i) {
+      if (dice.faces[i] == x)
+         sum += x;
+   }
+   return sum;
+}
+
+typedef struct {
+   uint8_t faces[6];
+} count_t;
+
+static count_t count(const dice_t dice)
+{
+   count_t counts = { 0 };
+   for (int i = 0; i < 5; ++i) {
+      assert(dice.faces[i] >= 1 && dice.faces[i] <= 6);
```

Comment:
I'm not sure that we've used assert() in an exercise before. Do you know what kind of output or results this shows the user when running the tests?

Normally we tend to try and output some error value, though I don't know if there is a good way to do that in this exercise?

Reply:
On other tracks it's assumed that inputs are not invalid. This isn't production so I like to try to keep these exercises focused and not worry about making them bulletproof against all types of misuse.

Reply:
This is not the test, but the example implementation. I'm using `i - 1` as an index further down, so this is also supposed to be sort of a comment reminding the reader that the faces go from 1 to 6, while we use indices 0 to 5 for them. I would find an `assert` good practice here and that's why I put it. If you insist, I can of course remove it.

Reply:
> Do you know what kind of output or results this shows the user when running the tests?

Just to be really clear, when a student does the exercise, they will not interact with this file in any way.